src/query/storages/stream/src/stream_table.rs

```diff
@@ -392,15 +430,45 @@ impl Table for StreamTable {
         let number_of_blocks = latest_summary
             .block_count
             .abs_diff(base_summary.block_count);
-
-        Ok(Some(TableStatistics {
+        let max_stats = TableStatistics {
             num_rows: Some(num_rows),
             data_size: Some(data_size),
             data_size_compressed: Some(data_size_compressed),
             index_size: Some(index_size),
             number_of_blocks: Some(number_of_blocks),
             number_of_segments: None,
-        }))
+        };
+        // In order to join order, the statistics are predicted, which may have a large bias.
+        let min_stats = TableStatistics {
+            num_rows: Some(num_rows / 2),
+            data_size: Some(data_size / 2),
+            data_size_compressed: Some(data_size_compressed / 2),
+            index_size: Some(index_size / 2),
+            number_of_blocks: Some(number_of_blocks / 2),
+            number_of_segments: None,
+        };
```

Comment:
how about making this calculation lazy? (since it may not be used later):

e.g.

let min_stats = || { TableStatistics ....
}


and alter, returns `Ok(Some(min_stats()))` if necessary